### PR TITLE
Feature/cicd

### PR DESCRIPTION
**Pull request by sebalopez**
Added GitHub Actions workflows

### CODE REVIEW

**Review by aguscha333 — APPROVED** [review]

**Review by jpirazusta — APPROVED** [review]

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  KEYS_BUCKET: ${{ secrets.AWS_S3_KEYS_BUCKET }}
+  BUILDS_BUCKET: ${{ secrets.AWS_S3_BUILDS_BUCKET }}
+  # iOS Release
+  APPLE_PROFILE: ${{ secrets.APPLE_DEV_PROFILE }}
+  APPLE_CERT: ios_distribution.cer
+  APPLE_KEY: Certificates.p12
+  APPLE_KEY_PASSWORD: ${{ secrets.APPLE_KEY_PASSWORD }}  
+  APPLE_TEAM_ID: ${{ secrets.APPLE_TEAM_ID}}
```
Comment (fernandatoledo) [inline .github/workflows/builds.yml:25]:
```suggestion
  FASTLANE_TEAM_ID: ${{ secrets.APPLE_TEAM_ID}}
```

It's later used in the Appfile and Fastfile with that name

File: .github/workflows/builds.yml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  FASTLANE_ITC_TEAM_ID: ${{ secrets.APPLE_TEAM_ITC_ID}}
```
Comment (fernandatoledo) [inline .github/workflows/builds.yml:26]: This is not being used, probably changed after the change with the .p8 key file

File: ios/fastlane/Fastfile [hunk 3]
```diff
@@ -17,26 +17,39 @@ default_platform(:ios)
...
-slack_url     = ENV["SLACK_URL"]                  # Slack webhook url and channel name for sending notifications upon completion
-slack_channel = ENV["SLACK_CHANNEL"]              #
+# REQUIRED ENVIRONMENT VARIABLES
+# ==============================
+# FASTLANE_USER                                 : Your App Store Connect / Apple Developer Portal id used for managing certificates and submitting to the App Store
+# FASTLANE_PASSWORD                             : Your App Store Connect / Apple Developer Portal password, usually only needed if you also set the FASTLANE_USER variable
+# MATCH_PASSWORD                                : Password used for match encryption
+# FASTLANE_TEAM_ID                              : Developer Portal team id
+# FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD  : Application specific password for FASTLANE_USER, required to upload a binary to App Store Connect
+# FASTLANE_ITC_TEAM_ID                          : App Store Connect team id, required for binary submission
```
Comment (fernandatoledo) [inline ios/fastlane/Fastfile:27]: If this is not being used, then we should also remove this line

File: .github/workflows/builds.yml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  FASTLANE_USER: ${{ secrets.FASTLANE_USER }}
+  FASTLANE_PASSWORD: ${{ secrets.FASTLANE_PASSWORD }}  
+  FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD: ${{ secrets.FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD }}
```
Comment (fernandatoledo) [inline .github/workflows/builds.yml:29]: These are not being used anymore as well

Comment (fernandatoledo) [inline .github/workflows/builds.yml:29]: The APP_STORE_CONNECT_API_KEY_FILE,
 APP_STORE_CONNECT_API_KEY_KEY_ID and 
 APP_STORE_CONNECT_API_KEY_ISSUER_ID 
 are missing

File: .github/workflows/builds.yml [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    - name: install gem dependencies
+      run: bundle update
+      working-directory: android
+    # Cleans workspace
+    # Runs build with Gradle
+    # Submits build to Play Store
+    # Sends Slack notification
+    - name: Build with Fastlane
+      run: bundle exec fastlane release_dev_appstore
```
Comment (fernandatoledo) [inline .github/workflows/builds.yml:118]: That lane does not exist
```suggestion
      run: bundle exec fastlane release_develop
```

**Review by fernandatoledo — CHANGES_REQUESTED** [review]

File: .github/workflows/builds.yml (hunk 5 above)
Comment (sebalopez) [reply .github/workflows/builds.yml]: thanks for all the catches!  happens when propagating changes across repos back and forth. Should all be fixed now

File: ios/fastlane/Appfile [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# app_identifier("[[APP_IDENTIFIER]]") # The bundle identifier of your app
+# apple_id("[[APPLE_ID]]") # Your Apple email address
```
Comment (fernandatoledo) [inline ios/fastlane/Appfile:2]:
```suggestion
# apple_id("[[FASTLANE_USER]]") # Your Apple email address
```

**Review by fernandatoledo — APPROVED** [review]

File: ios/fastlane/README.md [hunk 7]
```diff
@@ -17,66 +16,56 @@ xcode-select --install
...
+
+1. Generate certificate and profiles for each target
+2. [Disable automatic signing](https://developer.apple.com/library/archive/documentation/Security/Conceptual/CodeSigningGuide/Procedures/Procedures.html#//apple_ref/doc/uid/TP40005929-CH4-SW7) for each target in XCode and associate to the right provisioning profile
+3. For uploading the builds to TestFlight, [AppStore Connect API](https://developer.apple.com/documentation/appstoreconnectapi/creating_api_keys_for_app_store_connect_api) keys are required
+4. For uploading the builds to S3, [AWS keys](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys) with valid permissions are required
+
+# Required environment variables
+
+* `FASTLANE_USER`                         : Your App Store Connect / Apple Developer Portal id used for managing certificates and submitting to the App Store
+* `FASTLANE_PASSWORD`                     : Your App Store Connect / Apple Developer Portal password, usually only needed if you also set the
```
Comment (fernandatoledo) [inline ios/fastlane/README.md:29]: This is not being used anymore

File: ios/fastlane/Fastfile [hunk 8]
```diff
@@ -17,26 +17,37 @@ default_platform(:ios)
...
-s3_key        = ENV["AWS_ACCESS_KEY_ID"]          #
-s3_secret_key = ENV["AWS_SECRET_ACCESS_KEY"]      #
-s3_bucket     = ENV["BUILD_BUCKET"]               # AWS S3 bucket and folder to upload builds to (for Adhoc builds)
-folder        = ENV["FOLDER"]                     # 
-slack_url     = ENV["SLACK_URL"]                  # Slack webhook url and channel name for sending notifications upon completion
-slack_channel = ENV["SLACK_CHANNEL"]              #
+# REQUIRED ENVIRONMENT VARIABLES
+# ==============================
+# FASTLANE_USER                                 : Your App Store Connect / Apple Developer Portal id used for managing certificates and submitting to the App Store
+# FASTLANE_PASSWORD                             : Your App Store Connect / Apple Developer Portal password, usually only needed if you also set the FASTLANE_USER variable
```
Comment (fernandatoledo) [inline ios/fastlane/Fastfile:23]: This is not being used anymore, probably should be replaced with APP_STORE_CONNECT_API_KEY_FILE

**Review by fernandatoledo — APPROVED** [review]

**Pull request merged**